### PR TITLE
Remove code related to Google Analytics tracking

### PR DESCRIPTION
**Pull request by rafpaf**
This PR removes code relating to tracking events with Google Analytics

Comments with a TODO or a FIXME are meant to be discussed in code review, not merged.

### CODE REVIEW

File: frontend/src/metabase/admin/datamodel/table.js [hunk 1]
```diff
@@ -28,13 +23,9 @@ export const rescanTableFieldValues = createThunkAction(
 export const discardTableFieldValues = createThunkAction(
   DISCARD_TABLE_VALUES,
   function (tableId) {
-    return async function (dispatch, getState) {
+    return async function () {
```
Comment (rafpaf) [inline frontend/src/metabase/admin/datamodel/table.js:26]:
```suggestion
    return async function (dispatch, getState) {
```

File: frontend/src/metabase/admin/databases/database.js [hunk 2]
```diff
@@ -159,44 +157,24 @@ export const createDatabase = function (database) {
...
-        database.engine,
-      );
       throw error;
     }
   };
 };
 
 export const updateDatabase = function (database) {
-  return async function (dispatch, getState) {
+  return async function (dispatch) {
```
Comment (rafpaf) [inline frontend/src/metabase/admin/databases/database.js:170]:
```suggestion
  return async function (dispatch, getState) {
```

File: frontend/src/metabase/admin/databases/database.js [hunk 3]
```diff
@@ -217,16 +195,11 @@ export const saveDatabase = function (database) {
 };
 
 export const deleteDatabase = function (databaseId, isDetailView = true) {
-  return async function (dispatch, getState) {
+  return async function (dispatch) {
```
Comment (rafpaf) [inline frontend/src/metabase/admin/databases/database.js:198]:
```suggestion
  return async function (dispatch, getState) {
```

File: frontend/src/metabase/admin/datamodel/field.js [hunk 4]
```diff
@@ -10,13 +9,9 @@ export const DISCARD_FIELD_VALUES =
 export const rescanFieldValues = createThunkAction(
   RESCAN_FIELD_VALUES,
   function (fieldId) {
-    return async function (dispatch, getState) {
+    return async function () {
```
Comment (rafpaf) [inline frontend/src/metabase/admin/datamodel/field.js:12]:
```suggestion
    return async function (dispatch, getState) {
```

File: frontend/src/metabase/admin/datamodel/field.js [hunk 5]
```diff
@@ -28,13 +23,9 @@ export const rescanFieldValues = createThunkAction(
 export const discardFieldValues = createThunkAction(
   DISCARD_FIELD_VALUES,
   function (fieldId) {
-    return async function (dispatch, getState) {
+    return async function () {
```
Comment (rafpaf) [inline frontend/src/metabase/admin/datamodel/field.js:26]:
```suggestion
    return async function (dispatch, getState) {
```

File: frontend/src/metabase/setup/actions.ts [hunk 6]
```diff
@@ -62,14 +59,9 @@ export const loadDefaults = createAsyncThunk<void, void, ThunkConfig>(
   },
 );
 
-export const SELECT_STEP = "metabase/setup/SUBMIT_WELCOME_STEP";
+export const SELECT_STEP = "metabase/setup/SELECT_STEP";
```
Comment (rafpaf) [inline frontend/src/metabase/setup/actions.ts:62]: I'm assuming this was always supposed to be `"SELECT_STEP"`, not `"SELECT_WELCOME_STEP"` which was assigned to `SUBMIT_WELCOME` on the deleted line 68 below

File: frontend/src/metabase/setup/components/DatabaseStep/DatabaseStep.tsx [hunk 7]
```diff
@@ -102,7 +97,9 @@ export const DatabaseStep = (): JSX.Element => {
           <InviteUserForm
             user={user}
             invite={invite}
-            onSubmit={handleInviteSubmit}
+            // TODO: Since all this did was fire a Google Analytics event, can we remove the whole form?
+            // onSubmit={handleInviteSubmit}
+            onSubmit={() => null}
```
Comment (rafpaf) [inline frontend/src/metabase/setup/components/DatabaseStep/DatabaseStep.tsx:102]: Question for reviewers

File: frontend/src/metabase-lib/metadata/utils/tables.js [hunk 8]
```diff
@@ -1,7 +1,8 @@
 import getGAMetadata from "promise-loader?global!metabase-lib/metadata/utils/ga-metadata"; // eslint-disable-line import/default
 
 export async function injectTableMetadata(table) {
-  // HACK: inject GA metadata that we don't have intergrated on the backend yet
+  // HACK: inject GA metadata that we don't have integrated on the backend yet
+  // TODO: Remove GA stuff here?
```
Comment (rafpaf) [inline frontend/src/metabase-lib/metadata/utils/tables.js:5]: This is a question for reviewers

File: frontend/src/metabase/lib/settings.ts [hunk 9]
```diff
@@ -230,6 +230,7 @@ class MetabaseSettings {
     return !!(this.get("uploads-enabled") && this.get("uploads-database-id"));
   }
 
+  // TODO: Remove this code related to Google Analytics?
```
Comment (rafpaf) [inline frontend/src/metabase/lib/settings.ts:233]: This is a question for reviewers

File: frontend/src/metabase/setup/components/UserStep/UserStep.tsx [hunk 10]
```diff
@@ -57,7 +53,9 @@ export const UserStep = (): JSX.Element => {
       <UserForm
         user={user}
         onValidatePassword={validatePassword}
-        onSubmit={handleSubmit}
+        // TODO: Since all this did was fire a Google Analytics event, can we remove the whole form?
+        // onSubmit={handleSubmit}
+        onSubmit={() => null}
```
Comment (rafpaf) [inline frontend/src/metabase/setup/components/UserStep/UserStep.tsx:58]: Question for reviewers

**Comment by rafpaf** [comment]
I think a lot of tests are breaking because in my zeal I unnecessarily deleted some reducers.

This branch breaks the buttons on the right side of the page on the native query editor. It's surprising that this doesn't break any unit tests...

**Comment by npfitz** [comment]
Yeah, I spent a few minutes specifically looking into the errors on the setup tests. There are a few places where you've some `Thunks` because they basically just fired off events. The issue though is that those `Thunks` also emitted events for reducers. I think if you replace the `Thunks` with simeple `createAction(ACTION_NAME)` and restored the reducers, things might be a bit happier

**Comment by npretto** [comment]
We also have some backend code that should be ok to delete after this: https://github.com/metabase/metabase/blob/master/src/metabase/public_settings.clj#L234-L247 not sure how it works though, does it need a migration or similar?

**Pull request closed without merging**